torch/utils/tensorboard/writer.py

```diff
@@ -0,0 +1,714 @@
+"""Provides an API for generating Event protocol buffers."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import json
+import os
+import six
+import time
+
+from tensorboard.compat.proto.event_pb2 import SessionLog
+from tensorboard.compat.proto.event_pb2 import Event
+from tensorboard.compat.proto import event_pb2
+from tensorboard.summary.writer.event_file_writer import EventFileWriter
+
+from ._convert_np import make_np
+from ._embedding import make_mat, make_sprite, make_tsv, append_pbtxt
+from ._onnx_graph import load_onnx_graph
+from ._pytorch_graph import graph
+from ._utils import figure_to_image
+from .summary import (
+    scalar, histogram, histogram_raw, image, audio, text,
+    pr_curve, pr_curve_raw, video, custom_scalars, image_boxes
+)
+
+
+class FileWriter(object):
+    """Writes `Summary` protocol buffers to event files.
+    The `FileWriter` class provides a mechanism to create an event file in a
+    given directory and add summaries and events to it. The class updates the
+    file contents asynchronously. This allows a training program to call methods
+    to add data to the file directly from the training loop, without slowing down
+    training.
```

Comment:
When would I want to use this class? How do I use it?

Reply:
What exactly is an event? Via the API on this class, I can dump a graph. How is that an event?

Reply:
@ezyang can you see if my updates make sense? @lanpa might need your help here.